src/cargo/core/compiler/unit_dependencies.rs

```diff
@@ -353,6 +377,7 @@ fn compute_deps<'a, 'cfg>(
 /// the returned set of units must all be run before `unit` is run.
 fn compute_deps_custom_build<'a, 'cfg>(
     unit: &Unit<'a>,
+    dep_kind: DepKind,
```

Comment:
For the purpose of computing features, building the `build.rs` needs to use the features from the parent package.  If it always used "build", it would pick the wrong features for shared dependencies.  Example:

```
foo
├── foo build.rs (RunCustomBuild) NORMAL features
│   └── foo build.rs (Build) NORMAL features
│       └── common 
│           └── common build.rs (RunCustomBuild) BUILD-DEPENDENCY features
│               └── common build.rs (Build) BUILD-DEPENDENCY features
└── common
    └── common build.rs (RunCustomBuild) NORMAL features
        └── common build.rs (Build) NORMAL features
``` 

Here, when creating the `common build.rs` units, it wants the features based on its context (within a build dependency or not).  This is important for building build.rs because scripts often use cfg!() expressions for determining which features are enabled (as opposed to environment variables).